impl/maven-di/src/main/java/org/apache/maven/di/impl/InjectorImpl.java

```diff
@@ -220,7 +220,13 @@ public <Q> Supplier<Q> doGetCompiledBinding(Dependency<Q> dep) {
         if (key.getRawType() == List.class) {
             Set<Binding<Object>> res2 = getBindings(key.getTypeParameter(0));
             if (res2 != null) {
-                List<Supplier<Object>> list = res2.stream().map(this::compile).collect(Collectors.toList());
+                // Sort bindings by priority (highest first) for deterministic ordering
+                List<Binding<Object>> sortedBindings = new ArrayList<>(res2);
+                Comparator<Binding<Object>> comparing = Comparator.comparing(Binding::getPriority);
+                sortedBindings.sort(comparing.reversed());
```

Comment:
```suggestion
                sortedBindings.sort(Binding.getPriorityComparator());
```
could shift feature envy in dedicated domain `Binding`.

```java
public Comparator<Binding<Object>> getPriorityComparator() {
    return Comparator.comparing(Binding::getPriority);
}
```

Reply:
might do this, as leaner than implementing comparable. Both better then this to avoid exact this situation:

<img width="1155" alt="image" src="https://github.com/user-attachments/assets/28619530-9a60-4f42-b345-2492cd2c1968" />

DRY DRY DRY